### PR TITLE
[feat] Implement Idea Board #3

### PR DESCRIPTION
**Pull request by VulcanRaven737**
# Added the following

Idea Creation:
  * Button to Add Ideas:
  * Add a button labeled "Add Idea" on the page to add a new idea.
  
Idea Form:
  * Upon clicking the "Add Idea" button, display a form/modal where users can enter:
  * Title: A short, descriptive title for the idea.
  * Description: A brief description detailing the idea (optional).
  * Include a "Submit" button to save the new idea.
  * Validate the form inputs to ensure both the title and description are appropriately filled out before submission.
  
Display Ideas:
* The idea board page should consist of multiple ideas that constantly get updated.

All ideas need to display their:
  * Title
  * Description
  * Implement using [Cards](https://nextui.org/docs/components/card) for better visual representation.

Database Connection:
* Implement a connection to the database to store and retrieve ideas.
* Write SQl queries to create idea board table automatically
* Ideas Table:
** id: Primary key (UUID).
** title: VARCHAR for the idea title.
** description: TEXT for the idea description.
** created_at: TIMESTAMP for the submission date.
** Create Go structures for the idea board information fields.

Create new API endpoints:
* POST /create_idea: To create a new idea and save it to the database.
* GET /get_ideas: To fetch all submitted ideas from the database for display.

Fetch Ideas:
* When the idea board page loads, fetch existing ideas from the database and display them immediately.
* Implement proper error handling for fetching and creating ideas

### CODE REVIEW

**Review by shri30yans — CHANGES_REQUESTED** [review]
We are using mux for routing. Please use Mux instead.

File: Loop_backend/database/idea.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+// models/idea.go
+package models
+
+import (
+    "time"
+    "github.com/google/uuid"
+)
+
+type Idea struct {
```
Comment (shri30yans) [inline Loop_backend/database/idea.go:12]: Move structs to the `models.go` file.

File: Loop_backend/database/idea.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+func (h *IdeaHandler) CreateIdea(c *gin.Context) {
+    var req models.CreateIdeaRequest
+    if err := c.ShouldBindJSON(&req); err != nil {
+        c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request body"})
+        return
+    }
+
+    var idea models.Idea
```
Comment (shri30yans) [inline Loop_backend/database/idea.go]: Move SQL queries to the `queries.go` file.

File: Loop_backend/database/idea.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    "database/sql"
+    "log"
+    "your-project/handlers"
+    "github.com/gin-gonic/gin"
+    "github.com/lib/pq"
+)
+
+func main() {
+    // Database connection
+    db, err := sql.Open("postgres", "postgres://username:password@localhost:5432/dbname?sslmode=disable")
```
Comment (shri30yans) [inline Loop_backend/database/idea.go]: Please keep the Database files split like the original code.

File: Loop_frontend/app/ideaboard/actions.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+'use server'
+
+const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL;
+
+
+// Create a new project
+export async function createProject(projectData: any) {
```
Comment (shri30yans) [inline Loop_frontend/app/ideaboard/actions.ts:7]: `createProject` API route already exists. Migrate the method in the page of the ideaboard here.

File: Loop_frontend/app/ideaboard/page.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  const [errors, setErrors] = useState({ title: '', description: '' });
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  useEffect(() => {
+    fetchIdeas();
+  }, []);
+
+  const fetchIdeas = async () => {
```
Comment (shri30yans) [inline Loop_frontend/app/ideaboard/page.tsx:38]: Migrate function to actions.ts of this file.

File: Loop_backend/database/idea.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+    if err = db.Ping(); err != nil {
+        log.Fatal(err)
+    }
+
+    r := gin.Default()
+
+    ideaHandler := handlers.NewIdeaHandler(db)
+
+    r.POST("/api/ideas", ideaHandler.CreateIdea)
```
Comment (shri30yans) [inline Loop_backend/database/idea.go]: Do not create a new file for ideaBoard. Use the existing backend code and and API routes. Check the working of Projects page.

**Review by shri30yans — CHANGES_REQUESTED** [review]

**Comment by VulcanRaven737** [comment]
Hey can you award me partial points? I want to work on another issue for more points.

**Comment by shri30yans** [comment]
!bounty 70

**Comment by VulcanRaven737** [comment]
Thanks!

**Pull request closed without merging**